docs/src/main/asciidoc/includes/tracing/common-callbacks.adoc

```diff
@@ -0,0 +1,159 @@
+| `baggage()` | Returns the `Baggage` instance associated with the span.| &check;
+| `context()` | Returns the `SpanContext` associated with the span.| &check;
+| `status(Status)` | Sets the status of the span. | -
+| any `tag` method| Add a tag to the span. | &check;
+| `unwrap(Class)` | Cast the span to the specified implementation type. † | &check;
+
+|====
+
+{empty}* Helidon throws the link:{tracing-javadoc}/UnsupportedActivationException.html[`UnsupportedActivationException`] if a listener attempts an illegal operation from inside its `afterActivation` method. This Helidon exception extends `UnsupportedOperationException` and adds the `Scope scope()` method. Callers should catch this exception and close the `Scope`; Helidon will have activated the span and created the scope _before_ it invoked the listeners.
+
+† Helidon returns the unwrapped object, not a safe adapter to it.
+
+.link:{tracing-javadoc}/Scope.html[`io.helidon.tracing.Scope`] Operations
+|====
+| Method    | Purpose | OK?
+| `close()` | Close the scope. | -
+| `isClosed()`| Reports whether the scope is closed. | &check;
+|====
+
+.link:{tracing-javadoc}/SpanContext.html[`io.helidon.tracing.SpanContext`] Operations
+|====
+| Method    | Purpose | OK?
+| `asParent(Span.Builder)` | Sets this context as the parent of a new span builder. | &check;
+| `baggage()` | Returns `Baggage` instance associated with the span context. | &check;
+| `spanId()` | Returns the span ID. | &check;
+| `traceId()` | Returns the trace ID.| &check;
+
+|====
+
+=== Adding a Listener
+==== Explicitly Registering a Listener on a link:{tracing-javadoc}/Tracer.html[`Tracer`]
+Create a `{listener-type}` instance and invoke the `Tracer#register({listener-type})` method to make the listener known to that tracer.
+
+==== Automatically Registering a Listener on all `Tracer` Instances
+
+Helidon also uses Java service loading to locate listeners and register them automatically on all `Tracer` objects.
+Follow these steps to add an auto-loaded listener.
+
+1. Implement the link:{tracing-api-javadoc}/{listener-type}.html[`{listener-type}`] interface.
+2. Declare your implementation as a loadable service:
+   a. Create the file `META-INF/services/io.helidon.tracing.{listener-type}` containing a line with the fully-qualified name of your class which implements `{listener-type}`.
+   b. If your service has a `module-info.java` file add the following line to it:
++
+[source,java,subs="attributes+"]
+----
+provides io.helidon.tracing.{listener-type} with <your-implementation-class>;
+----
+
+The `{listener-type}` interface declares default no-op implementations for all the methods, so your listener can implement only the methods it needs to.
+
+Helidon invokes each listener's methods in the following order:
+
+|====
+| Method | When invoked
+
+| `beforeStart(Span.Builder<?> spanBuilder)` | Just before a span is started from its builder.
+| `afterStart(Span span)` | Just after a span has started.
```

Comment:
Perhaps consider, in Helidon-flavored verbiage: `afterX` ▶️ `xed`

Reply:
Will change.